pwndbg/gdblib/disasm/__init__.py

```diff
@@ -120,6 +120,10 @@ def clear_on_reg_mem_change() -> None:
     lambda: None
 )
 
+# Maps an address to integer 0/1, indicating the Thumb mode bit for the given address.
+# Value is None if Thumb bit irrelevent or unknown.
+emulated_arm_mode_cache: DefaultDict[int, int] = collections.defaultdict(lambda: None)
```

Comment:
This cache is never cleared. Is that ok?

Also can't you have a single address once executed as Thumb and once as ARM mode? Like a polyglot ARM & Thumb. I think u can?

Reply:
I think we can safely not clear this cache. In the same way the code has used `backward_cache`, the values will always be updated/overridden by more recent data when we step near the instructions again. We 'renew' the cache values for the next future instructions every time we step, so for the case of a polyglot binary, the cache will contain the values from the most recent time we stepped near the instructions in question. The cache values will always be correct in the current context (whether we are interpreting the instructions as Arm/Thumb at this moment in the polyglot binary).

They could be stale (if we step at the instructions and they are non-Thumb, then we don't step at the location for a while, and instead hopped between some breakpoints that are somewhere else, and the polyglot binary in between has switched to Thumb, and then we do `nearpc address_that_was_non_thumb_when_i_stepped_there`), but I think this is so incredibly rare that it's a better experience to just keep the cache always uncleared, so when someone does `nearpc` at some address they've been at, it will get the correct thumb/arm mode transitions.

Reply:
Shall we maybe clear it at least on program start or attach (probably newobjfile event)?

I mean: if we never clear it and it maps address->0|1, then having a super long GDB session where u restart an ASLRed binary 100 times will make this cache bigger and bigger, effectively leaking memory.